### PR TITLE
refactor: api

### LINKED ISSUE #83: Move application logic out of main package
As I understand it, writing a Golang utility that includes Tanka isn't possible (in an elegant way) because it doesn't expose an API.

It would be very useful if the commands were separated out into separate packages available by import into other go code.

### PR DESCRIPTION
**Pull request by sh0rez**
Refactors all of the actual application logic out of the main package, into the `github.com/grafana/tanka/pkg/tanka` package, which effectively forms an API for Tanka, allowing it to be imported into other projects and used from there. 

The API is loosely based on the CLI, ie it follows the same scheme:
```go
tanka.Apply("environments/default`)
```

To change the behaviour of these calls, for example the diff strategy, modifiers can be used (https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis):

```go
tanka.Apply("environments/default", tanka.WithApplyForce(true))
```

In brief testing it seems as if it is still working the same way as it did before, however everyone reviewing this might want to keep a close look at the changes to see if I accidentally broke something without noticing. Thanks :)

Resolves #83

### CODE REVIEW

File: cmd/tk/env.go [hunk 1]
```diff
@@ -50,7 +51,7 @@ func envSetCmd() *cobra.Command {
 	envSettingsFlags(&tmp, cmd.Flags())
 
 	name := cmd.Flags().String("name", "", "")
-	cmd.Flags().MarkHidden("name")
+	_ = cmd.Flags().MarkHidden("name")
```
Comment (rfratto) [inline cmd/tk/env.go]: nit: can we remove the redundant assign here?

File: cmd/tk/workflow.go [hunk 2]
```diff
@@ -88,58 +72,31 @@ func diffCmd() *cobra.Command {
...
+			h := highlight("diff", *changes)
 			pageln(h)
 		} else {
-			fmt.Println(h)
+			fmt.Println(*changes)
 		}
-	} else {
-		fmt.Println(*changes)
+
+		os.Exit(16)
```
Comment (rfratto) [inline cmd/tk/workflow.go]: Can the `16` be brought out to a package-level constant so this doesn't feel as magic number-y?

File: cmd/tk/env.go (hunk 1 above)
Comment (sh0rez) [reply cmd/tk/env.go]: it returns an error, the linter is unhappy about it. This was intended to singal that we don't care about the error

Comment (rfratto) [reply cmd/tk/env.go]: Got it 👍

File: pkg/tanka/tanka.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	// target regular expressions to limit the working set
+	targets []*regexp.Regexp
+
+	// additional options for diff
+	diff kubernetes.DiffOpts
+	// additional options for apply
+	apply kubernetes.ApplyOpts
+}
+
+// Modifiers allow to
```
Comment (rfratto) [inline pkg/tanka/tanka.go]: Incomplete comment?

File: pkg/tanka/tanka.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
+// The tanka package allows to use most of Tankas features available on the
```
Comment (rfratto) [inline pkg/tanka/tanka.go]:
```suggestion
// The tanka package allows to use most of Tanka's features available on the
```

Extremely minor nit, sorry

File: cmd/tk/main.go [hunk 5]
```diff
@@ -21,47 +20,19 @@ import (
...
 	interactive = terminal.IsTerminal(int(os.Stdout.Fd()))
 )
 
-// list of deprecated config keys and their alternatives
-// however, they still work and are aliased internally
-var deprecated = map[string]string{
-	"namespace": "spec.namespace",
-	"server":    "spec.apiServer",
-	"team":      "metadata.labels.team",
-}
```
Comment (rfratto) [inline cmd/tk/main.go]: Was this just dead code?

**Pull request merged**